Aquality.Selenium/src/Aquality.Selenium/Browsers/RemoteBrowserFactory.cs

```diff
@@ -18,12 +19,23 @@ public override Browser Browser
         {
             get
             {
+                AqualityServices.LocalizedLogger.Info("loc.browser.grid");
                 var browserProfile = AqualityServices.Get<IBrowserProfile>();
                 var capabilities = browserProfile.DriverSettings.DriverOptions.ToCapabilities();
                 var timeoutConfiguration = AqualityServices.Get<ITimeoutConfiguration>();
                 var driver = AqualityServices.Get<IActionRetrier>().DoWithRetry(() => 
-                    (RemoteWebDriver)Activator.CreateInstance(typeof(RemoteWebDriver), 
-                    browserProfile.RemoteConnectionUrl, capabilities, timeoutConfiguration.Command));
+                {
+                    try
+                    {
+                        return new RemoteWebDriver(browserProfile.RemoteConnectionUrl, capabilities, timeoutConfiguration.Command);
+                    }
+                    catch (Exception e)
+                    {
+                        AqualityServices.LocalizedLogger.Fatal("loc.browser.grid.fail", e);
+                        throw;
+                    }                    
+                }, new[] { typeof(WebDriverException) });
```

Comment:
Maybe move it to separate private method?

Reply:
reworked